tsdb/chunks/chunks.go

```diff
@@ -190,40 +190,50 @@ func (w *Writer) cut() error {
 		return err
 	}
 
-	p, _, err := nextSequenceFile(w.dirFile.Name())
+	n, f, _, err := cutSegmentFile(w.dirFile, chunksFormatV1, true, w.segmentSize)
 	if err != nil {
 		return err
 	}
+	w.n = int64(n)
+
+	w.files = append(w.files, f)
+	if w.wbuf != nil {
+		w.wbuf.Reset(f)
+	} else {
+		w.wbuf = bufio.NewWriterSize(f, 8*1024*1024)
+	}
+
+	return nil
+}
+
+func cutSegmentFile(dirFile *os.File, chunksFormat byte, preallocate bool, segmentSize int64) (headerSize int, newFile *os.File, seq int, err error) {
```

Comment:
I would avoid boolean arguments to functions. It's not extensible and can be confusing. I think in this case you can just not preallocate if `segmentSize` is 0, right?